src/components/QuranReader/EndOfScrollingControls/index.tsx

```diff
@@ -1,59 +1,36 @@
 import React from 'react';
 
+import ChapterControls from './ChapterControls';
 import styles from './EndOfScrollingControls.module.scss';
+import JuzControls from './JuzControls';
+import PageControls from './PageControls';
+import TafsirControls from './TafsirControls';
+import VerseControls from './VerseControls';
 
-import Button, { ButtonType } from 'src/components/dls/Button/Button';
 import { QuranReaderDataType } from 'src/components/QuranReader/types';
-import { getChapterData, isFirstSurah, isLastSurah } from 'src/utils/chapter';
-import { VersesResponse } from 'types/ApiResponses';
+import Verse from 'types/Verse';
 
 interface Props {
   quranReaderDataType: QuranReaderDataType;
-  initialData: VersesResponse;
+  verses: Verse[];
 }
 
-const EndOfScrollingControls: React.FC<Props> = ({ quranReaderDataType, initialData }) => {
-  const isVerse = quranReaderDataType === QuranReaderDataType.Verse;
-  // only show the controls for the verse for now.
-  if (!isVerse) {
-    return <></>;
-  }
-  const [firstVerse] = initialData.verses;
-  const { chapterId, verseNumber } = firstVerse;
-  const { versesCount } = getChapterData(String(chapterId));
-  const chapterNumber = Number(chapterId);
-  const isLastVerseOfSurah = verseNumber === versesCount;
-
+const EndOfScrollingControls: React.FC<Props> = ({ quranReaderDataType, verses }) => {
+  const lastVerse = verses[verses.length - 1];
```

Comment:
nit, not sure if it's possible to only pass the last verse from the parent, instead of passing the entire verses array — only to extract the last verse.